lib/web/apiserver.go

```diff
@@ -360,6 +360,10 @@ func NewHandler(cfg Config, opts ...HandlerOption) (*APIHandler, error) {
 	h.GET("/webapi/sites/:site/nodes/:server/:login/scp", h.WithClusterAuth(h.transferFile))
 	h.POST("/webapi/sites/:site/nodes/:server/:login/scp", h.WithClusterAuth(h.transferFile))
 
+	// Sign required files to setup mTLS in other services (eg DBs)
+	// GET /webapi/sites/:site/sign?hostname=<hostname>&ttl=<ttl>&format=<format>
+	h.GET("/webapi/sites/:site/sign", h.WithProvisionTokenAuth(h.signCertKeyPair))
```

Comment:
As `WithProvisionTokenAuth` invalidates the token, this call is not [idempotent](https://developer.mozilla.org/en-US/docs/Glossary/Idempotent) nor [safe](https://developer.mozilla.org/en-US/docs/Glossary/Safe/HTTP) as `GET` requests are supposed to. The `POST` method might be more appropriate here.

Reply:
Yeah, I was unsure about which verb to use
I ended up using `GET` because:
- using `POST` usually means creating a new instance of a given resource, something you can `GET` afterwards by some `<resource>/<id>` - we don't have such resource currently
- the request is idempotent and safe **if** we ignore the authentication method
  - that's a big if, but if we were using some other type of authentication, it could as well expire after some time (or some usage - things like rate limiting)
- the issue suggested a `GET` (of course it can be changed if required :+1: )

No strong opinion here, but my tendency would be to use this GET instead of a POST that doesn't create a new entity.

Reply:
Another issue can be caching, I don't know if teleport sets anti-cache headers by default (I hope so :crossed_fingers: ).

Don't bother changing before we have @timothyb89 's opinion.